.github/workflows/container-images.yml

```diff
@@ -0,0 +1,139 @@
+---
+name: Container images
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - .github/workflows/container-images.yml
+      - .github/workflows/path-filter/container-images.yml
+      - container-images/**
+
+permissions: read-all
+
+jobs:
+  detect-changes:
+    name: Detect changes
+    runs-on: ubuntu-latest
+    outputs:
+      images: ${{ steps.detect_changes.outputs.changes }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Detect changes
+        id: detect_changes
+        uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # v2.11.1
+        with:
+          filters: .github/workflows/path-filter/container-images.yml
+
+  build:
+    if: ${{ needs.detect-changes.outputs.images != '[]' }}
+    needs: [detect-changes]
+    name: Build image
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      packages: write
+    strategy:
+      matrix:
+        images: ${{ fromJson(needs.detect-changes.outputs.images) }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Install cosign
+        id: install_cosign
+        uses: sigstore/cosign-installer@c3667d99424e7e6047999fb6246c0da843953c65 # v3.0.1
+
+      - name: Prepare build paramerters
+        id: prepare
+        run: |
+          echo "name=$(jq -r '.name' container-images/${{ matrix.images }}/config.json)" >> ${GITHUB_OUTPUT}
```

Comment:
I think you can use github_env to pass values between steps. Github_outputs is for moving things between jobs.